539_group_feedings_required

cookeac opened the pull request:
Add required fields to `icarGroupFeedingEventResource`.
Needs `eventDateTime`, `location`, `groupMethod` to make sense and align with other group events.

Resolves #539

cookeac committed:
Add required fields to icarGroupFeedingEventResource

AndreasSchultzGEA commented on resources/icarGroupFeedingEventResource.json:
```diff
@@ -1,6 +1,7 @@
 {
     "description": "Event for recording group or mob feeding. Allowances represent averages so this cannot be used to populate individual animal events.",
     "type": "object",
+    "required": ["eventDateTime", "location", "groupMethod"],
```
"groupMethod" is also a required property in the parent.
Unsure, if this repetition is needed

AlexeyHardCode replied:
@AndreasSchultzGEA good finding! it seems this "groupMethod" is already required, so, no need to make it reqired in the icarGroupFeedingEventResource.

cookeac replied:
I was also unsure if we needed to add `groupMethod` again when adding the other required properties. If code generation will merge the required lists, then we don't need to add it again. However, it does help with readability to be able to look at the concrete object and see all required fields.

cookeac committed:
Remove groupMethod from required

Remove groupMethod from required - it IS still required, but is already specified in icarGroupEventResource.
